push-subscription-management/server.js

```diff
@@ -0,0 +1,61 @@
+var webPush = require('web-push');
+// collection of endpoints stored in memory (in production it should be
+// a database)
+var subscriptions = [];
```

Comment:
Could we link the subscription to a particular user? Instead of storing only the endpoints, you could store a user ID and its endpoint. This way when the subscription changes and the client notifies the server, the server can overwrite the user's endpoint.